topology.go

```diff
@@ -156,7 +164,51 @@ func QueryTopology(dbpool *pgxpool.Pool, params TopologyOptions) ([]*models.Comp
 	// If ID is present, we do not apply any filters to the root component
 	results = applyStatusFilter(results, params.ID != "", params.Status...)
 
-	return results, nil
+	var res TopologyResponse
+	res.Components = results
+	populateTopologyResult(results, &res)
+
+	res.Teams, err = GetTeamNamesOfComponents(ctx, dbpool, res.componentIDs)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get teams of components: %w", err)
+	}
+
+	return &res, nil
+}
+
+func GetTeamNamesOfComponents(ctx context.Context, dbpool *pgxpool.Pool, componentIDs []string) ([]string, error) {
+	if len(componentIDs) == 0 {
+		return nil, nil
+	}
+
+	placeholders := make([]string, len(componentIDs))
+	for i := range componentIDs {
+		placeholders[i] = fmt.Sprintf("$%d", i+1)
+	}
+
+	args := make([]any, len(componentIDs))
+	for i, id := range componentIDs {
+		args[i] = id
+	}
```

Comment:
would much prefer to use gorm instead of doing this primitive stuff here but the `QueryTopology` func is implemented to work with `pgxpool.Pool`. 

Let me know if change it so that it accepts `gorm.DB` instead.